docs/advanced/mixin/5.overwrite.md

```diff
@@ -0,0 +1,53 @@
+@Overwrite
+=========
+:::caution
+This is ```@Overwrite``` but NOT ```@Override```.
+:::
+
+Overwriting is a technology that discards all codes in a vanilla method and replaces it with your own code. It is a thorough way to change the vanilla mechanisms.
+
+:::danger
+It can be dangerous to completely change vanilla mechanisms.
+:::
+
+You should not consider overwriting unless:
+
+- You want to completely change vanilla mechanisms.
+- You want to remove some codes from these vanilla methods.
+- You want to fix the bugs in these vanilla methods.
+- You have a better implementation than these vanilla codes.
+
+You can write overwriting methods in your mixin class, which will be applied to the target class.
+
+Overwriting methods are annotated with ```@Overwrite``` and they must have the same names, parameter types, and sequences as the vanilla methods. What can be changed is only the parameter names. Otherwise, the overwriting will not take effect.
+
+Here is an example of an overwriting method
+
+```java
+@Overwrite
+public void methodName(Type1 param1, Type2 param2){
+	...
+}
+```
+
+whose target is
+```java
+public void methodName(Type1 param1, Type2 param2){
+	...
+}
+
+```
+
+Once you have overwritten some method, you should add the JavaDoc, including the author's name and reasons, to the method.
+```java
+/**
+* @author
+* @reason
+*/
+@Overwrite
+...
+```
```

Comment:
`@reason` is not a valid javadoc tag by default. Also, you should always document your code regardless.